### PR TITLE
[PIL-735] - Incomplete data check

### PR DESCRIPTION
**Pull request by jassalrichy**

### CODE REVIEW

**Review by alikouhestani — APPROVED** [review]

**Review by alikouhestani — APPROVED** [review]

**Review by TevynAllen — APPROVED** [review]

**Review by alikouhestani — APPROVED** [review]

File: test/controllers/fm/NfmEntityTypeControllerSpec.scala [hunk 1]
```diff
@@ -77,6 +77,8 @@ class NfmEntityTypeControllerSpec extends SpecBase {
         ).toString
       }
     }
+
+    // TODO - This test will not work
```
Comment (MarjanMubarok) [inline test/controllers/fm/NfmEntityTypeControllerSpec.scala:81]: Is this TODO still relevant?

File: app/helpers/SubscriptionHelpers.scala [hunk 2]
```diff
@@ -103,31 +103,43 @@ trait SubscriptionHelpers {
...
-        ((get(SubAddSecondaryContactPage).contains(true) & secondaryTelephone) | get(SubAddSecondaryContactPage).contains(false))
-        & get(SubRegisteredAddressPage).isDefined
-    ) true
-    else false
-
   def finalStatusCheck: Boolean =
-    if (groupDetailStatus == RowStatus.Completed & fmStatus == RowStatus.Completed & upeStatus == RowStatus.Completed & groupDetailStatusChecker)
-      true
-    else false
+    if (
```
Comment (MarjanMubarok) [inline app/helpers/SubscriptionHelpers.scala:136]: You can remove the if here and as long as the & is at the end of each boolean line it should compile fine.

File: test/controllers/fm/NfmEntityTypeControllerSpec.scala (hunk 1 above)
Comment (jassalrichy) [reply test/controllers/fm/NfmEntityTypeControllerSpec.scala:81]: Yes, need to look at refactoring this test. It's not part of this ticket hence for futrure ref.

File: app/helpers/SubscriptionHelpers.scala (hunk 2 above)
Comment (jassalrichy) [reply app/helpers/SubscriptionHelpers.scala]: Done

File: app/controllers/subscription/GroupDetailCheckYourAnswersController.scala [hunk 3]
```diff
@@ -50,7 +50,7 @@ class GroupDetailCheckYourAnswersController @Inject() (
     if (request.userAnswers.groupDetailStatus == RowStatus.Completed) {
       Ok(view(list))
     } else {
-      Redirect(controllers.routes.JourneyRecoveryController.onPageLoad())
+      Redirect(controllers.subscription.routes.InprogressTaskListController.onPageLoad)
```
Comment (TevynAllen) [inline app/controllers/subscription/GroupDetailCheckYourAnswersController.scala:53]: Any tests for this?

File: app/controllers/subscription/ContactCheckYourAnswersController.scala [hunk 4]
```diff
@@ -66,18 +66,16 @@ class ContactCheckYourAnswersController @Inject() (
...
     }
   }
 
-  def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
-    for {
-      updatedAnswers <-
-        Future.fromTry(request.userAnswers.set(CheckYourAnswersLogicPage, true))
-      _ <- userAnswersConnectors.save(updatedAnswers.id, Json.toJson(updatedAnswers.data))
-    } yield Redirect(controllers.routes.TaskListController.onPageLoad)
+  def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/controllers/subscription/ContactCheckYourAnswersController.scala:78]: Done

File: app/controllers/subscription/GroupDetailCheckYourAnswersController.scala (hunk 3 above)
Comment (jassalrichy) [reply app/controllers/subscription/GroupDetailCheckYourAnswersController.scala:53]: Done

File: app/controllers/CheckYourAnswersController.scala [hunk 5]
```diff
@@ -55,17 +57,28 @@ class CheckYourAnswersController @Inject() (
...
+          .getOrElse(
+            Ok(
+              view(upeSummaryList, nfmSummaryList, groupDetailSummaryList, primaryContactSummaryList, secondaryContactSummaryList, addressSummaryList)
+            )
+          )
+      }
     }
   }
 
+  private def setCheckYourAnswersLogic(userAnswers: UserAnswers)(implicit hc: HeaderCarrier): Future[UserAnswers] =
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/controllers/CheckYourAnswersController.scala:75]: Done

**Review by TevynAllen — APPROVED** [review]

**Pull request merged**